src/core/net/ip6.cpp

```diff
@@ -1058,42 +1057,36 @@ otError Ip6::ProcessReceiveCallback(const Message &    aMessage,
         case kProtoUdp:
         {
             Udp::Header udp;
+            uint16_t    destPort;
+
             aMessage.Read(aMessage.GetOffset(), sizeof(udp), &udp);
 
-            switch (udp.GetDestinationPort())
-            {
-            case Mle::kUdpPort:
+            destPort = udp.GetDestinationPort();
 
+            if (destPort == Mle::kUdpPort)
+            {
                 // do not pass MLE messages
                 if (aMessageInfo.GetSockAddr().IsLinkLocal() || aMessageInfo.GetSockAddr().IsLinkLocalMulticast())
                 {
                     ExitNow(error = OT_ERROR_NO_ROUTE);
                 }
-
-                break;
-
-#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE == 0
-            case kCoapUdpPort:
-
+            }
+#if !OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
+            if (destPort == kCoapUdpPort)
```

Comment:
Not using `else-if` here is also intentional?

Reply:
Not really, we can use `else` here. 
I just changed all of them to `if (dstPort == {something})` but we can keep the `else` here. Will change it.

Reply:
Changed this in new pushed commit. Aslo combined the indented two `if`s into one single `if` check (for each case).